### PR TITLE
Small fixes reported by coverity

### PR DESCRIPTION
**Pull request by rgacogne**
I would appreciate any comment, especially regarding the NSEC3*RecordContent changes. The removed fields don't seem to be used, but I'm really not familiar with this code and thus might very well be missing something.

### CODE REVIEW

File: pdns/syncres.cc [hunk 1]
```diff
@@ -283,6 +283,9 @@ bool SyncRes::doOOBResolve(const DNSName &qname, const QType &qtype, vector<DNSR
 void SyncRes::doEDNSDumpAndClose(int fd)
 {
   FILE* fp=fdopen(fd, "w");
+  if (!fp) {
+    return;
```
Comment (cmouse) [inline pdns/syncres.cc]: Should this cause some error/warning?

**Comment by Habbie** [comment]
NSEC3 code looks good. We already use the 'int len' trick for NSEC3RecordContent and this syncs the PARAM code with that.

File: pdns/syncres.cc (hunk 1 above)
Comment (rgacogne) [inline pdns/syncres.cc]: You are right, but the error should get to the right channel, so we would need to return a error to the caller or move the `fdopen()` to the calling function. Similar functions (`MemRecursorCache::doDumpNSSpeeds()`, `MemRecursorCache::doDump()`, `RecursorPacketCache::doDump()`, ...) don't seem to care.

Comment (cmouse) [inline pdns/syncres.cc]: :ok:

**Pull request merged**